merino/providers/top_picks.py

```diff
@@ -0,0 +1,157 @@
+"""Top Pick Navigational Queries Provider"""
+import asyncio
+import json
+import logging
+import os
+from collections import defaultdict
+from typing import Any, Optional
+
+from fastapi import FastAPI
+
+from merino.config import settings
+from merino.providers.base import BaseProvider, BaseSuggestion
+
+SCORE: float = settings.providers.top_picks.score
+LOCAL_TOP_PICKS_FILE: str = settings.providers.top_picks.top_picks_file_path
+QUERY_CHAR_LIMIT: int = settings.providers.top_picks.query_char_limit
+
+
+logger = logging.getLogger(__name__)
+
+
+class Suggestion(BaseSuggestion):
+    """Model for Top Pick Query Suggestion"""
+
+    block_id: int
+    is_top_pick: bool
+
+
+class Provider(BaseProvider):
+    """Top Pick Query Suggestion Provider"""
+
+    # In normal usage this is None, but tests can create the provider with a
+    # FastAPI instance to fetch mock responses from it. See `__init__()`.
+    _app: Optional[FastAPI]
+
+    primary_index: defaultdict = defaultdict(list)
+    secondary_index: defaultdict = defaultdict(list)
+    results: list[dict[str, Any]]
```

Comment:
Noticed a few type mismatches as well, I'm surprised mypy didn't catch any of them?

```suggestion
    results: list[Suggestion]
```